base/dict.jl

```diff
@@ -258,11 +258,12 @@ function empty!(h::Dict{K,V}) where V where K
 end
 
 # get the index where a key is stored, or -1 if not present
-function ht_keyindex(h::Dict{K,V}, key) where V where K
+@assume_effects :terminates_locally function ht_keyindex(h::Dict{K,V}, key) where V where K
     isempty(h) && return -1
     sz = length(h.keys)
     iter = 0
     maxprobe = h.maxprobe
+    maxprobe < sz || error() # This error will never trigger, but is needed for terminates_locally to be valid
```

Comment:
```suggestion
    maxprobe < sz || error("UNREACHABLE") # This error will never trigger, but is needed for terminates_locally to be valid
```

Reply:
I didn't want to allocate a string. is that fear unfounded?

Reply:
`error()` does not exist as a method. the string allocation would be in the branch.

Reply:
alternative `throw(AssertionError())`